src/monitor/ManualAgentsValidator.cpp

```diff
@@ -0,0 +1,21 @@
+
+#include "ManualAgentsValidator.hpp"
+#include "AgentInformation.hpp"
+
+
+void ManualAgentsValidator::addNewAgent(const QString& name, const QString& ip, const QString& port)
+{
+    const auto segments = ip.split('.');
+    const bool ipValid = std::all_of(segments.begin(), segments.end(), [](const QString& segment) {
+        const int number = segment.toInt();
+        return number >= 0 && number <= 255;
+    });
+
+    const int portNum = port.toInt();
+
+    if (ipValid && portNum >= 0 && portNum <= 65535)
+    {
+        AgentInformation info(name, ip, portNum, AgentInformation::DetectionSource::Hardcoded);
```

Comment:
Maybe before adding procedure monitor should ask aged is it up and running? GET message with "status" or something?